wrapper.go

```diff
@@ -50,7 +68,36 @@ type Wrapper interface {
 	Decrypt(context.Context, *EncryptedBlobInfo, []byte) ([]byte, error)
 }
 
+// LifecycleWrapper is a Wrapper that implements lifecycle management for keys in a KMS.
+type LifecycleWrapper interface {
+	Wrapper
+
+	// ImportKey creates a named key by importing key material in the given KeyEntry.
+	// The key will have the given KeyType, Purpose, and ProtectionLevel if supported by the implementation.
+	// Returns the ID of a new key version and an error.
+	ImportKey(ctx context.Context, name string, kt keysutil.KeyType, ke keysutil.KeyEntry, pr Purpose, pl ProtectionLevel) (string, error)
+
+	// RotateKey rotates the named key by creating a new key version with key material in the given KeyEntry.
+	// The key version will have the given KeyType, Purpose, and ProtectionLevel if supported by the implementation.
+	// Returns the ID of a new key version and an error.
+	RotateKey(ctx context.Context, name string, kt keysutil.KeyType, ke keysutil.KeyEntry, pr Purpose, pl ProtectionLevel) (string, error)
```

Comment:
The args list looks a little long for ImportKey and RotateKey, I wonder if passing a struct here would make sense instead. Just a thought.

Reply:
I agree. I thought about passing a struct for these as well. Happy to change this to use a struct if others agree!

Reply:
Depending on how the implementation consumes `KeyEntry`, this might also be an opportunity to remove the dependency on the `keysutil` package. The struct could contain all the information needed for the operation, such as the key type, key bytes, purpose, protection level, etc.

Reply:
I like the idea of removing the `keysutil` package dependency. I've changed `ImportKey` and `RotateKey` to take in a newly defined struct, `KMSKey`, which contains all of the information necessary for those operations. Please have a look at let me know your thoughts.